locksmith/src/graphql/datasource/keysToRenew.ts

```diff
@@ -0,0 +1,56 @@
+import { gql } from 'apollo-server-express'
+import { GraphQLDataSource } from 'apollo-datasource-graphql'
+import networks from '@unlock-protocol/networks'
+
+export class KeysToRenew extends GraphQLDataSource {
+  async getKeysToRenew(
+    start: number,
+    end: number,
+    network: number,
+    page: number
+  ): Promise<any[]> {
+    this.baseURL = networks[network].subgraphURI
+    const keysToRenewQuery = gql`
+      query Keys($start: Int!, $end: Int!, $skip: Int) {
+        keys(
+          skip: $skip
+          orderBy: expiration
+          orderDirection: desc
+          where: { expiration_gte: $start, expiration_lte: $end }
```

Comment:
@searchableguy do you know if it is possible to do use subquery in a where clause in graphQL? 

I would like to do sth like (as only Lock after v10 support recurring memberships)

```gql
where: { 
   expiration_gte: $start 
   expiration_lte: $end 
   lock {
    version_gte: 10
  }
}
```

Reply:
Nope. The current subgraph doesn't have any such filter for the lock version on keys. Your best bet filtering the keys with the v10 lock version or query all the v10 locks, then add all the keys array together.